proxy-libs/heliospg/heliospg_test.go

```diff
@@ -0,0 +1,110 @@
+package heliospg
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/go-pg/pg/v10/orm"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type User struct {
+    Id     int64
+    Name   string
+    Emails []string
+}
+
+// createSchema creates database schema for User and Story models.
+func createSchema(db *DB) error {
+    models := []interface{}{
+        (*User)(nil),
+    }
+
+    for _, model := range models {
+        err := db.Model(model).CreateTable(&orm.CreateTableOptions{
+            Temp: true,
+        })
+        if err != nil {
+            return err
+        }
+    }
+    return nil
+}
+
+func getSpanRecorder() *tracetest.SpanRecorder {
+	spanRecorder := tracetest.NewSpanRecorder()
+	otel.SetTracerProvider(sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(spanRecorder)))
+	propagator := propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{})
+	otel.SetTextMapPropagator(propagator)
+	return spanRecorder
+}
+
+func ConnectToDb() *DB {
+	db := Connect(&Options{
+        User: "postgres",
+    })
+	defer db.Close()
+	err := createSchema(db)
+    if err != nil {
+        panic(err)
+    }
+	return db
+}
+
+func insertUser(db *DB, id int, name string, role string) {
+	user1 := &User{
+        Name:   "admin",
+        Emails: []string{"admin1@admin", "admin2@admin"},
+    }
+    _, err := db.Model(user1).Insert()
+    if err != nil {
+        panic(err)
+    }
+}
+
+func assertSpan(t *testing.T, spanRecorder *tracetest.SpanRecorder) []attribute.KeyValue {
+	spanRecorder.ForceFlush(context.Background())
+	spans := spanRecorder.Ended()
+	assert.Equal(t, 1, len(spans))
+	span := spans[0]
+	// assert.Equal(t, "users.insert", span.Name())
+	assert.Equal(t, trace.SpanKindClient, span.SpanKind())
+	return span.Attributes()
+}
+
+func assertAttributes(t *testing.T, attributes []attribute.KeyValue, id int, name string, role string) {
+	for _, attribute := range attributes {
```

Comment:
Please use the assert.contains syntax (e.g., `assert.Contains(t, attributes, attribute.String("db.name", "test"))`)

Reply:
The problem with the iterations is that if the key isn't there, it'll still succeed